samples/StatusCodePagesSample/Startup.cs

```diff
@@ -25,53 +25,67 @@ public void Configure(IApplicationBuilder app)
             // app.UseStatusCodePagesWithReExecute("/errors/{0}");
 
             // "/[?statuscode=400]"
-            app.Use((context, next) =>
+            app.Use(async (context, next) =>
             {
-                if (context.Request.Path.HasValue && !context.Request.Path.Equals(new PathString("/")))
-                {
-                    return next();
-                }
-
                 // Check for ?statuscode=400
                 var requestedStatusCode = context.Request.Query["statuscode"];
                 if (!string.IsNullOrEmpty(requestedStatusCode))
                 {
                     context.Response.StatusCode = int.Parse(requestedStatusCode);
-                    return Task.FromResult(0);
-                }
 
-                var builder = new StringBuilder();
-                builder.AppendLine("<html><body>");
-                builder.AppendLine("<a href=\"" +
-                    WebUtility.HtmlEncode(context.Request.PathBase.ToString()) + "/missingpage/\">" +
-                    WebUtility.HtmlEncode(context.Request.PathBase.ToString()) + "/missingpage/</a><br>");
+                    // To turn off the StatusCode feature - For example the below code turns off the StatusCode middleware 
+                    // for even status codes for demonstration.
+                    if (context.Response.StatusCode % 2 == 0)
```

Comment:
Because the HTML page generated with status codes is long with `200` entries it would be hard if I just disable it on one status code. Alternative status code did not look random to me as there is a pattern. Alternatively I can generate links in the HTML page for both status code pages enabled as well as disabled. How does that look?